src/backend/InvenTree/build/test_build.py

```diff
@@ -722,6 +724,57 @@ def test_new_build_notification(self):
 
         self.assertTrue(messages.filter(user__pk=4).exists())
 
+    def test_events(self):
+        """Test that build events are triggered correctly."""
+
+        from django_q.models import OrmQ
+        from build.events import BuildEvents
+
+        settings.TESTING_TABLE_EVENTS = True
+        settings.PLUGIN_TESTING_EVENTS = True
+        set_global_setting('ENABLE_PLUGINS_EVENTS', True)
+
+        OrmQ.objects.all().delete()
+
+        # Create a new build
+        build = Build.objects.create(
+            reference='BO-9999',
+            title='Some new build',
+            part=self.assembly,
+            quantity=5,
+            issued_by=get_user_model().objects.get(pk=2),
+            responsible=Owner.create(obj=Group.objects.get(pk=3))
+        )
+
+        # Check that the 'build.created' event was triggered
+        task = findOffloadedEvent(
+            'build_build.created',
+            matching_kwargs=['id', 'model'],
+            reverse=True,
+            clear_after=True,
+        )
+
+        # Assert that the task was found
+        self.assertIsNotNone(task)
+
+        # Check that the Build ID matches
+        self.assertEqual(task.kwargs()['id'], build.pk)
+
+        # Issue the build
+        build.issue_build()
+
+        # Check that the 'build.issued' event was triggered
+        task = findOffloadedEvent(
+            BuildEvents.ISSUED,
+            matching_kwargs=['id'],
+            clear_after=True,
+        )
+
+        self.assertIsNotNone(task)
+
+        settings.TESTING_TABLE_EVENTS = False
+        settings.PLUGIN_TESTING_EVENTS = False
```

Comment:
This teardown could not be run if there are issues with the asserts above and cause contagion with other tests.

Reply:
Thanks, I'll fix this